python_modules/dagster/dagster/core/definitions/asset_layer.py

```diff
@@ -101,6 +101,43 @@ def _resolve_input_to_destinations(
     return all_destinations
 
 
+def _resolve_output_to_destinations(output_name, asset_key, node_def, handle):
```

Comment:
I think this function can potentially be made simpler. We already have the inner_node_handle on line 508, so we might be able to immediately resolve this to a set of downstream solid inputs using graph_def.dependency_structure.output_to_downstream_inputs_for_solid. If there's not a way to get that to work, ignore me.

also the function is named a little confusingly. It might be best to actually just respect the name of the function and return a list of node input handles, then map those node input handles to the output asset key in the outer function.

Reply:
Using the `graph_def.dependency_structure.output_to_downstream_inputs_for_solid` will only work if `graph_def` is the immediate graph node nesting the outputting solid, it will not work in the cases that there are multiple graphs nested. In those cases, we will still need to traverse every single nested graph to see if any graph's output is passed as an input to an op/graph at the same level.

Reply:
I will rename the function and modify it to just return the list of node input handles